pandas/core/reshape/merge.py

```diff
@@ -1798,6 +1799,24 @@ def _get_merge_keys(self):
 
         return left_join_keys, right_join_keys, join_names
 
+    def _maybe_coerce_merge_keys(self):
```

Comment:
why is this not just a part of _validate_specification? seems like re-inventing the wheel here

Reply:
Because this is way easier after knowing the join keys than before. But I think we could probably move this into validate_specification

Reply:
yeah would really like it to be part of the main check